packages/client/src/data_commons_client.ts

```diff
@@ -0,0 +1,352 @@
+          if (typeof geometryString !== "string") {
+            return false;
+          }
+          return true;
+        })
+        .map((dataRow) => {
+          const geometryString = dataRow[`entity.${geoJsonProperty}`] as string;
+          const geometry = JSON.parse(geometryString) as Geometry;
+          const properties = { ...dataRow };
+          delete properties[`entity.${geoJsonProperty}`];
+          const feature: Feature = {
+            geometry,
+            properties,
+            type: "Feature",
+          };
+          if (feature.geometry && shouldRewind) {
+            return rewind(feature, { reverse: true });
+          }
+          return feature;
+        }),
+      type: "FeatureCollection",
+    };
+    return geoJson;
+  }
+
+  /**
+   * Fetches data commons variable values about an entity or entities.
+   * @param params
+   * @returns
+   */
+  async getDataRows(params: GetDataRowsParams): Promise<DataRow[]> {
+    // Fetch variable observations
+    const pointApiResponse =
+      "parentEntity" in params
+        ? await this.webClient.getObservationsPointWithin(params)
+        : await this.webClient.getObservationsPoint(params);
+    if (!pointApiResponse) {
+      return [];
+    }
+    const entityDcids =
+      this.getEntityDcidsFromPointApiResponse(pointApiResponse);
+    // Fetch relevant entity and variable property values
+    const entityPropValues = await this.getNodePropValues(
+      entityDcids,
+      params.entityProps || DEFAULT_ENTITY_PROPS
+    );
+    const variablePropValues = await this.getNodePropValues(
+      params.variables,
+      params.variableProps || DEFAULT_VARIABLE_PROPS
+    );
+
+    // Fetch population data for per capita calculations
+    let populationObservations: SeriesApiResponse = { data: {}, facets: {} };
+    if (!_.isEmpty(params.perCapitaVariables)) {
+      populationObservations =
+        "parentEntity" in params
+          ? await this.webClient.getObservationsSeriesWithin({
+              ...params,
+              variables: [TOTAL_POPULATION_VARIABLE],
+            })
+          : await this.webClient.getObservationsSeries({
+              ...params,
+              variables: [TOTAL_POPULATION_VARIABLE],
+            });
+    }
+
+    const dataRows = this.getDataRowsFromObservations(
+      entityDcids,
+      params.variables,
+      pointApiResponse,
+      entityPropValues,
+      variablePropValues,
+      populationObservations
+    );
+
+    return Promise.resolve(dataRows);
+  }
+
+  /**
+   * Fetches the first node property value for the given property name
+   * @param params
+   * @returns
+   */
+  async getFirstNodeValues(params: {
+    dcids: string[];
+    prop: string;
+  }): Promise<Record<string, string | null>> {
+    const nodePropvals = await this.webClient.getNodePropvals(params);
+    const nodeValues: Record<string, string | null> = {};
+    Object.keys(nodePropvals).forEach((variableDcid) => {
```

Comment:
nit: nodePropvals aren't keyed by variable dcids, are they? maybe change `variableDcid` to something that describes the keys

Reply:
Looks like [it is keyed by dcid](https://datacommons.org/api/node/propvals/out?dcids=geoId/18&dcids=Count_Person&prop=name), but it could be any node dcid (variable or entity). Updated to `nodeDcid`